src/BarrelPlanarMPGDTracker_geo.cpp

```diff
@@ -283,14 +291,20 @@ static Ref_t create_BarrelPlanarMPGDTracker_geo(Detector& description, xml_h e,
         string module_name = _toString(module, "module%d");
         DetElement mod_elt(lay_elt, module_name, module);
         double mod_z       = 0.5 * dimensions.length();
-        double z_placement = mod_z - j * nz * mod_z; // z location for module placement
+        double z_placement = mod_z - 0.5 * pcb_feb_ext -
+                             j * (nz * mod_z - pcb_feb_ext); // z location for module placement
         double z_offset =
             z_placement > 0
                 ? -z0 / 2.0
                 : z0 / 2.0; // determine the amount of overlap in z the z nz modules have
 
-        Transform3D tr(RotationZYX(0.0, ((M_PI / 2) - phic - phi_tilt), -M_PI / 2),
-                       Position(xc, yc, mpgd_pos.z() + z_placement + z_offset)); // in x-y plane,
+        Transform3D tr(
+            RotationZYX(0, ((M_PI / 2) - phic - phi_tilt), -M_PI / 2) * RotationZ(j * M_PI),
+            Position(
+                xc, yc,
+                mpgd_pos.z() + z_placement +
+                    z_offset)); //RotZYX rotat planes around azimuth, RotZ flip plane so pcb_feb_ext is faceing endcaps
```

Comment:
```suggestion
        Transform3D tr(
            // RotationZYX rotates planes around azimuth, RotZ flips plane so pcb_feb_ext is facing endcaps
            RotationZYX(0, ((M_PI / 2) - phic - phi_tilt), -M_PI / 2) * RotationZ(j * M_PI),
            Position(
                xc, yc,
                mpgd_pos.z() + z_placement +
                    z_offset));
```
Needs spelling corrections.